events_task/test/my_app_web/controllers/confirmation_controller_test.exs

```diff
@@ -0,0 +1,73 @@
+defmodule MyApp.ConfirmationControllerTest do
+  use MyAppWeb.ConnCase
+  alias MyApp.Accounts
+  alias MyApp.Content
+
+  @valid_event_attrs %{
+    "description" => "TestText",
+    "timestamp" => %{
+      "day" => "1",
+      "hour" => "0",
+      "minute" => "0",
+      "month" => "1",
+      "year" => "2015"
+    }
+  }
+
+  describe "confirmations" do
+    def create_test_user(attrs \\ %{}) do
+      {:ok, user} =
+        attrs
+        |> Enum.into(%{email: "sergei@gmail.com"})
+        |> Accounts.create_user()
+
+      user
+    end
+
+    def create_test_event(attrs \\ %{}) do
+      {:ok, event} =
+        attrs
+        |> Enum.into(@valid_event_attrs)
+        |> Content.create_event()
+
+      event
+    end
+
+    test "successfully create confirmation with valid params" do
+      event = create_test_event()
+      user = create_test_user()
+
+      conn =
+        build_conn()
+        |> post(confirmation_path(conn, :create, %{"event_id" => event.id, "user_id" => user.id}))
+
+      assert conn.private[:plug_session]["phoenix_flash"] == %{
+               "info" => "Confirmed successfully."
+             }
+    end
+
+    test "fails double confirmation" do
+      event = create_test_event()
+      user = create_test_user()
+
+      conn =
+        build_conn()
+        |> post(confirmation_path(conn, :create, %{"event_id" => event.id, "user_id" => user.id}))
+
+      assert conn.private[:plug_session]["phoenix_flash"] == %{
+               "info" => "Confirmed successfully."
+             }
+
+      conn =
+        post(
+          conn,
+          confirmation_path(conn, :create, %{"event_id" => event.id, "user_id" => user.id})
+        )
+
+      assert conn.private[:plug_session]["phoenix_flash"] == %{
```

Comment:
You can clear up old flash messages before next request using [clear_flash/1](https://hexdocs.pm/phoenix/Phoenix.ConnTest.html#clear_flash/1)